src/main/kotlin/org/dif/crypto/key/RecipientKeySelector.kt

```diff
@@ -21,4 +24,55 @@ class RecipientKeySelector(private val didDocResolver: DIDDocResolver, private v
             didDoc.findVerificationMethod(verificationMethodId)
         }
     )
+
+    fun authCryptKeys(fromSelector: String, toSelector: List<String>): Pair<Key, List<Key>> {
+        val (did) = divideDIDFragment(fromSelector)
+        val didDoc = didDocResolver.resolve(did).orElseThrow { throw DIDDocNotResolvedException(did) }
+
+        return if (isDIDFragment(fromSelector)) {
+            val sender = didDocResolver.resolve(did)
+                .map { it.findVerificationMethod(fromSelector) }
+                .map { Key.wrapVerificationMethod(it) }
+                .orElseThrow { throw SecretNotFoundException(fromSelector) }
+
+            val recipients = findRecipientKeys(toSelector, sender.curve)
+            Pair(sender, recipients)
+        } else {
+            didDoc.keyAgreements
+                .asSequence()
+                .map { didDoc.findVerificationMethod(it) }
+                .map { Key.wrapVerificationMethod(it) }
+                .map { Pair(it, findRecipientKeys(toSelector, it.curve)) }
+                .find { it.second.isNotEmpty() }
+                ?: throw IncompatibleCryptoException("DID Doc does not contain compatible verification method")
+        }
+    }
+
+    fun anonCryptKeys(selector: List<String>): List<Key> = findRecipientKeys(selector, null)
+        .ifEmpty { throw IncompatibleCryptoException("DID Doc does not contain compatible verification method") }
+
+    private fun findRecipientKeys(selector: List<String>, curve: Curve?): List<Key> = selector.flatMap {
+        didUrl ->
+        val (did) = divideDIDFragment(didUrl)
+        val didDoc = didDocResolver.resolve(did).orElseThrow { throw DIDDocNotResolvedException(did) }
```

Comment:
No need to call potentially expensive operation `didDocResolver.resolve(did)` if  `isDIDFragment(didUrl)`